src/components/container/HeartEssentialEight/HeartEssentialEight.tsx

```diff
@@ -0,0 +1,215 @@
+import iconLoseWeight from "./icons/loseweight.svg";
+import iconManageBloodPressure from "./icons/managebloodpressure.svg";
+import iconReduceBloodSugar from "./icons/reducebloodsugar.svg";
+import iconSleep from "./icons/sleep.svg";
+import iconStopSmoking from "./icons/stopsmoking.svg";
+
+export type HeartEssentialEightScores =
+  | "Activity"
+  | "BloodPressure"
+  | "BloodSugar"
+  | "BMI"
+  | "Cholesterol"
+  | "Diet"
+  | "Nicotine"
+  | "Sleep";
+
+export interface HeartEssentialEightProps {
+  previewState?: "Default" | "NoData";
+  customFieldDictionary: Record<HeartEssentialEightScores, string> & {
+    [key: string]: string;
+  };
+  innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export default function (props: HeartEssentialEightProps) {
+  const customRingStyle: React.CSSProperties = {
+    backgroundColor: "lightblue",
+    height: "100px",
+    width: "100px",
+  };
+
+  let [score, setScore] = React.useState<number>();
+  let [achievementMap, setAchievementMap] =
+    useState<Record<HeartEssentialEightScores, boolean>>();
+
+  async function initialize() {
+    if (props.previewState) {
+      if (["Default", "NoData"].includes(props.previewState ?? "")) {
+        const data =
+          props.previewState === "Default" ? previewHeartEssentialData : {};
+        loadScoresFromCustomFieldData(data);
+      }
+    } else {
+      MyDataHelps.getParticipantInfo().then((participantInfo) => {
+        loadScoresFromCustomFieldData(participantInfo.customFields);
+      });
+    }
+  }
+
+  function loadScoresFromCustomFieldData(
+    customFieldData: Record<string, string>
+  ) {
+    let newScore: number = 0;
+    let newAchievementMap = {} as Record<HeartEssentialEightScores, boolean>;
+
+    for (const key in props.customFieldDictionary) {
+      if (props.customFieldDictionary[key]) {
+        let surveyScore = customFieldData[props.customFieldDictionary[key]];
+        let numericScore = parseInt(surveyScore);
+        if (!Number.isNaN(numericScore)) {
+          newScore += numericScore;
+          newAchievementMap[key as HeartEssentialEightScores] =
+            numericScore >= 100;
+        } else {
+          newAchievementMap[key as HeartEssentialEightScores] = false;
+        }
+      }
+    }
+
+    if (newScore > 0) {
+      newScore = (newScore / 800) * 100;
+    }
+
+    setScore(parseFloat(newScore.toFixed(1)));
+    setAchievementMap(newAchievementMap);
+  }
+
+  function generateAchievementMap(
+    didAchieve: boolean,
+    newAchievementMap: Record<HeartEssentialEightScores, boolean>
+  ) {
+    const map = newAchievementMap as Record<string, boolean>;
+    const achievementList: any[] = [];
+    for (const key in map) {
+      const vv = map[key];
+      if (vv === didAchieve) {
+        const title = key.replace(/([A-Z])/g, " $1").replace(/^./, (str) => {
+          return str.toUpperCase();
+        });
+        let icon: any;
+        let subtitle: any;
+        let url: string;
```

Comment:
It doesn't look like the URL variable ever gets set.  Are we still deciding what to do with these actions when clicked?